hardware/is_camera/src/camera.cc

```diff
@@ -306,19 +337,66 @@ namespace is_camera {
         bayer_img_msg_buffer_[i]->step = kImageWidth;
         bayer_img_msg_buffer_[i]->data.resize(kImageWidth * kImageHeight);
       }
-  }
-  // Enable or disable the feature timer
-  bool CameraNodelet::EnableBayerService(ff_msgs::SetBool::Request & req,
-                     ff_msgs::SetBool::Response & res) {
-    if (req.enable && !bayer_enable_) {
-      EnableBayer();
-    } else if (!req.enable && bayer_enable_) {
+    } else if (!enable && bayer_enable_) {
       bayer_pub_.shutdown();
+    } else {
+      return;
+    }
+    enable = bayer_enable_;
+  }
+
+  // Timer that periodically changes camera exposure based on the captured image's histogram
+  void CameraNodelet::AutoExposure() {
+    // If images not being captured, do not adjust exposure
+    if (!thread_running_ || ((pub_.getNumSubscribers() == 0) && (getNumBayerSubscribers() == 0))) return;
+
+    // Get last generated image from buffer (index incremented after image completed)
+    sensor_msgs::ImagePtr& grey_image = img_msg_buffer_[(img_msg_buffer_idx_ - 1) % kImageMsgBuffer];
+
+    cv::Mat input(kImageHeight, kImageWidth,
+                     cv::DataType<uint8_t>::type,
+                     &(grey_image->data[0]),
+                     kImageWidth);
+
+    // Calculate Histogram
+    cv::Mat hist;
+    int histSize = 5;
```

Comment:
I think what you actually want is to have histSize in the config file. Then desired_msv would be histSize / 2. (?)